pkg/build/controller/build/build_controller.go

```diff
@@ -1055,6 +1055,13 @@ func (bc *BuildController) resolveImageReferences(build *buildv1.Build, update *
 		// If we cannot resolve the output reference, the output image stream
 		// may not yet exist. The build should remain in the new state and show the
 		// reason that it is still in the new state.
+
+		// However, if source type is binary, it means, that user is creating it via
+		// the 'oc start-build'. in that case it makes sense to fail the build to let
+		// the user know early that resource does not exist
+		if build.Spec.Source.Type == buildv1.BuildSourceBinary {
+			update.setPhase(buildv1.BuildPhaseFailed)
+		}
```

Comment:
I am not sure about it, because this does not exit the code, it just makes sure build fails in case of `Binary` build and errors regarding output. Comments in this part of code suggest that not failing the build here is intended behavior in order to create `imagestream` or fix the error while build is hanging.

However, maybe it's not a good idea to let the build hang, and just fail build in case of any input type regardless?

Reply:
Per live discussion - this is fine. `transitionToPhase` doesn't give us a lot of extra benefit, and failing fast here lets the binary build end.